Engine_Revit_UI/Modify/SetParameter.cs

```diff
@@ -133,9 +133,33 @@ public static bool SetParameter(this Element element, BuiltInParameter builtInPa
 
         public static bool SetParameter(this Parameter parameter, object value, Document document = null)
         {
+            // Skip null and read-only parameters
             if (parameter == null || parameter.IsReadOnly)
                 return false;
 
+            // Workset parameters
+            // Filter for worksets
+            if (parameter.Id.IntegerValue == (int)BuiltInParameter.ELEM_PARTITION_PARAM)
+            {
+                // Find an existing workset with a specified name if it exists
+                string worksetName = value as string;
+                Workset workset = Query.Workset(document, worksetName);
```

Comment:
As mentioned in the other comment, all `public static` methods in BHoM should be extension methods, so this could look as follows (and is preferred):

```suggestion
                Workset workset = document.Workset(worksetName);
```